applications/ChimeraApplication/custom_processes/rotate_region_process.h

```diff
@@ -0,0 +1,260 @@
+ * the modelpart accordingly.
+ * @author Aditya Ghantasala
+ */
+class KRATOS_API(CHIMERA_APPLICATION) RotateRegionProcess : public Process {
+public:
+  /// Pointer definition of MoveRotorProcess
+  KRATOS_CLASS_POINTER_DEFINITION(RotateRegionProcess);
+
+  typedef ModelPart::NodeIterator NodeIteratorType;
+  typedef ProcessInfo ProcessInfoType;
+  typedef ProcessInfo::Pointer ProcessInfoPointerType;
+  typedef Matrix MatrixType;
+  typedef Vector VectorType;
+
+  /// Constructor.
+  RotateRegionProcess(ModelPart &rModelPart, Parameters rParameters);
+
+  /// Destructor.
+  virtual ~RotateRegionProcess();
+
+  void ExecuteBeforeSolutionLoop() override;
+
+  void SetAngularVelocity(const double NewAngularVelocity);
+
+  void SetTorque(const double NewTorque);
+
+  void ExecuteInitializeSolutionStep() override;
+
+  void ExecuteFinalizeSolutionStep() override;
+
+  void ExecuteAfterOutputStep() override;
+
+  virtual std::string Info() const override;
+
+  /// Print information about this object.
+  virtual void PrintInfo(std::ostream &rOStream) const override;
+
+  /// Print object's data.
+  void PrintData();
+
+protected:
+  ///@name Protected static Member Variables
+  ///@{
+
+  ///@}
+  ///@name Protected member Variables
+  ///@{
+
+  ///@}
+
+private:
+  ///@name Private static Member Variables
+  ///@{
+
+  ///@}
+  ///@name Private member Variables
+  ///@{
+
+  /**
+   * @class RotationSystem
+   * @ingroup KratosChimeraApplication
+   * @brief This class does the solve of Euler's equations of rotation around
+   * the given axis.
+   * @details Uses BDF2 time integration for calculating the current theta and
+   * omega
+   * @author Aditya Ghantasala
+   */
+  class RotationSystem {
+  public:
+    /// Pointer definition of RotationSystem
+    KRATOS_CLASS_POINTER_DEFINITION(RotationSystem);
+    /*
+     * @brief Constructor
+     * @param MomentOfInertia The moment of inertia of the system.
+     * @param DampingCoefficient The Damping Coefficient of the system.
+     */
+    RotationSystem(const double MomentOfInertia,
+                   const double DampingCoefficient = 0.0);
+
+    /*
+     * @brief Advances the rotation system in time
+     * @param NewTime Time where the system is to be set
+     */
+    void CloneTimeStep(const double NewTime, const double Dt);
+
+    /*
+     * @brief Applies the given torque on the system
+     * @param Torque The torque to be applied
+     */
+    void inline ApplyTorque(const double Torque);
+
+    /*
+     * @brief Calculates the current state of the system
+     */
+    double CalculateCurrentRotationState();
+    /*
+     * @brief Gives the current angle of rotation Theta
+     * @out Current angle of rotation Theta
+     */
+    double GetCurrentTheta() const;
+
+    /*
+     * @brief Gives the current angular velocity Omega
+     * @out Current angular velocity Omega
+     */
+    double GetCurrentOmega() const;
+
+  private:
```

Comment:
Can you briefly describe the member variables? There are quite a few...

Reply:
done !